packages/flame/lib/src/game/game.dart

```diff
@@ -355,11 +358,16 @@ abstract mixin class Game {
     gameStateListeners.remove(callback);
   }
 
+  bool _isInternalRefresh = false;
+
   /// When a Game is attached to a `GameWidget`, this method will force that
   /// widget to be rebuilt. This can be used when updating any property which is
-  /// implemented within the Flutter tree.
+  /// implemented within the Flutter tree.  [internalRefresh] when passed as
+  /// false, will trigger the `onAttach` and `onDetach` events; otherwise, those
+  /// events will not be called.
```

Comment:
```suggestion
  /// implemented within the Flutter tree.
  ///
  /// When [isInternalRefresh] is passed as false it will trigger the `onAttach`
  /// and `onDetach` events; otherwise, those events will not be called.
```